code-challenges.test.js

```diff
@@ -27,41 +27,88 @@ const numbersArray2 = [24, 27, 30, 33, 36]
 
-const object1 = { number: 15 }
+// describe ("divByThree", () => {
+//     it("returns a string letting you know if the entered number is devisible by 3 or not.", () => { 
+//         const object1 = { number: 15 }
+//         const object2 = { number: 0 }
+//         const object3 = { number: -7 } 
+//     expect(divByThree(object1.number)).toEqual(`${object1.number} is divisible by three`)
+//     expect(divByThree(object3.number)).toEqual(`${object3.number} is not divisible by three`)
+//     })
+// })
+//Good failur: ReferenceError: divByThree is not defined
 // Expected output: "15 is divisible by three"
-const object2 = { number: 0 }
 // Expected output: "0 is divisible by three"
-const object3 = { number: -7 }
 // Expected output: "-7 is not divisible by three"
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+// Pseudo code: Make a function that takes in a number. add a conditional with a sum to see if the entered number divided by 3 eventually = 0. have 2 if/else statements, if num / 3 eventually = 0, return "num (in string interpolation) is divisable by 3". If not return the opposite.
+
+// const divByThree = (number) => {
+//     if (number % 3 === 0) {
+//     return `${number} is divisible by three`
+//     }
+//     else {
+//     return `${number} is not divisible by three`
+//     } 
+// }
+
+// console.log(divByThree(number))
 
 // --------------------2) Create a function that takes in an array of words and returns an array with all the words capitalized.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+// describe("capNouns", () => {
+//     it("takes in an array and returns the same array capitalised", () => {
+//         const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
+//         const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
+//         expect(capNouns(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+//         expect(capNouns(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+//     })
+// })
+
+// Good fauilure:  ReferenceError: capNouns is not defined
+
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
-const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
 
 // b) Create the function that makes the test pass.
 
-// Pseudo code:
+// Pseudo code: write a function that takes in the array. use .map to iterate through the array, create an action, below return the value 0 indexed, this will access the first index of each value in the array. Use .toUpperCase to captialise the first index letter in the array, then concat the rest of the remaining word them with + (also .substring(1)) .
```

Comment:
You are accessing the 0 index of each value in the array, which would be a string, not the array itself.